crdt/sequence.py

```diff
@@ -0,0 +1,247 @@
+        Inserts an item to the sequence at the specified position. The position is a
+        0-based index from the perspective of the caller (e.g., does not count deleted
+        objects). This makes it easier for the caller to make insertions because the
+        caller does not have to keep track of object IDs.
+        """
+
+        # Tick the clock
+        self.clock.add(1)
+
+        index = self.index_transform(position)
+        target = self.sequence[index].operation
+        op_id = self.clock.get()
+
+        # Add the insert operation to the log and update the sequence
+        op = Operation(OpId(self.id, op_id), OperationType.INSERT, target, payload=object)
+        self.operations.add(op)
+        op.do(self.sequence)
+
+    def remove(self, position):
+        """
+        Removes an item from the set at the specified position. The position must be
+        within bounds of the current sequence, or this raises an IndexError.
+        """
+        # Tick the clock
+        self.clock.add(1)
+
+        index = self.index_transform(position)
+        target = self.sequence[index].operation
+        op_id = self.clock.get()
+
+        # Add the remove operation to the log and update the sequence
+        op = Operation(OpId(self.id, op_id), OperationType.REMOVE, target)
+        self.operations.add(op)
+        op.do(self.sequence)
+
+    def merge(self, other):
+        """
+        Merges another Sequence with this one.
+        """
+        if not isinstance(other, Sequence):
+            raise ValueError("Incompatible CRDT for merge(), expected Sequence")
+        
+        # Sync the local clock with the remote clock
+        self.clock = self.clock.merge(other.clock)
+
+        # Get the new operations to be applied
+        patch_ops = other.operations.get().difference(self.operations.get())
+        
+        # Get a sorted view of the patches to apply
+        patch_log = sorted(patch_ops, key=cmp_to_key(self.compare_operations))
+
+        # Path the sequence using the new operations
+        for op in patch_log:
+            op.do(self.sequence)
+
+        # Merge the two operation logs
+        self.operations = self.operations.merge(other.operations)
+        return self
+
+    def get(self):
+        """
+        Returns the sorted list of payloads in the sequence.
+        """
+        return [obj.operation.payload for obj in self.sequence if not obj.tombstone]
+
+class Object():
+    """
+    An Object represents a single item in a Sequence.
+    """
+    
+    def __init__(self, operation):
+        self.operation = operation
+        self.tombstone = False
+
+    def print(self):
+        """
+        Prints the object to stdout.
+        """
+        print("owner: {}, payload: {}, tombstone: {}".format(self.operation.owner.print(), self.operation.payload, self.tombstone))
+
+class OpId():
+    """
+    An OpId represents a unique identifier for an Operation.
+    """
+
+    def __init__(self, name, id):
```

Comment:
I recommend adding a docstring to explain what types `name` and `id` are and how they should be used, particularly since they are both used to determine precedence, but differently.